Tests/MeiliSearchUnitTests/DocumentsTests.swift

```diff
@@ -217,34 +217,111 @@ class DocumentsTests: XCTestCase {
     self.wait(for: [expectation], timeout: TESTS_TIME_OUT)
   }
 
-  func testGetDocuments() {
+  func testGetDocumentWithSparseFieldsets() {
     let jsonString = """
-      [{
+      {
         "id": 25684,
-        "release_date": "2020-04-04T19:59:49.259572Z",
-        "poster": "https://image.tmdb.org/t/p/w1280/iuAQVI4mvjI83wnirpD8GVNRVuY.jpg",
-        "title": "American Ninja 5",
-        "overview": "When a scientists daughter is kidnapped, American Ninja, attempts to find her, but this time he teams up with a youngster he has trained in the ways of the ninja."
-      },{
-        "id": 468219,
-        "title": "Dead in a Week (Or Your Money Back)",
-        "release_date": "2020-04-04T19:59:49.259572Z",
-        "poster": "https://image.tmdb.org/t/p/w1280/f4ANVEuEaGy2oP5M0Y2P1dwxUNn.jpg",
-        "overview": "William has failed to kill himself so many times that he outsources his suicide to aging assassin Leslie. But with the contract signed and death assured within a week (or his money back), William suddenly discovers reasons to live... However Leslie is under pressure from his boss to make sure the contract is completed."
-      }]
+        "title": "American Ninja 5"
+      }
+      """
+
+    // Prepare the mock server
+    session.pushData(jsonString, code: 200)
+
+    do {
+      let decoder = JSONDecoder()
+      decoder.dateDecodingStrategy = .formatted(Formatter.iso8601)
+      let data = jsonString.data(using: .utf8)!
+      let stubMovie: Movie = try decoder.decode(Movie.self, from: data)
+      let identifier: String = "25684"
+
+      // Start the test with the mocked server
+      let expectation = XCTestExpectation(description: "Get Movies document")
+
+      self.index.getDocument(identifier, fields: ["title", "id"]) { (result: Result<Movie, Swift.Error>) in
```

Comment:
Since your documents already only contain `title` and `id` this tests is not actually testing if `fields` is useful. Maybe only try by adding `id` or `title` to see if one of the fields is removed

Reply:
I didn't get it, because this is a unit test, so I control how the body will be retrieved, can you explain more this situation??

Reply:
Yep I did that in other test :)